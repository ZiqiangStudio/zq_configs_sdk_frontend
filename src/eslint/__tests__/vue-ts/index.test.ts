import { curryExecEslint } from '../util';

describe('Vue TS Lint', () => {
  const execEslint = curryExecEslint('vue-ts');

  test('报错', async () => {
    const { output, error } = await execEslint('error.vue');
    expect(error).toBeTruthy();
    expect(output).toMatchSnapshot();
  });

  test('正常通过', async () => {
    const { output, error } = await execEslint('normal.vue');
    expect(error).toBeFalsy();
    expect(output).toMatchSnapshot();
  });
});
