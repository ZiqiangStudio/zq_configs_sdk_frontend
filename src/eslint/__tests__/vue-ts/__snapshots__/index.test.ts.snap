// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vue TS Lint 报错 1`] = `" /vue-ts/error.vue    2:1   error  Insert \`··\`                                                                prettier/prettier    3:1   error  Replace \`··<p·class="title">{title}·\` with \`····<p·class="title">{title}\`  prettier/prettier    4:3   error  Insert \`··\`                                                                prettier/prettier    5:1   error  Insert \`··\`                                                                prettier/prettier    6:1   error  Insert \`··\`                                                                prettier/prettier    9:1   error  Delete \`··\`                                                                prettier/prettier    9:3   error  'onMount' is not defined                                                   no-undef   15:8   error  Must use \`.value\` to read or write the value wrapped by \`ref()\`            vue/no-ref-as-operand   20:2   error  Insert \`;\`                                                                 prettier/prettier   24:9   error  Replace \`'/api/users/'·+·id\` with \`('/api/users/'·+·id)\`                   prettier/prettier   24:31  error  Unexpected any. Specify a different type                                   @typescript-eslint/no-explicit-any   29:1   error  Delete \`⏎\`                                                                 prettier/prettier   33:18  error  Insert \`;\`                                                                 prettier/prettier  ✖ 13 problems (13 errors, 0 warnings)   11 errors and 0 warnings potentially fixable with the \`--fix\` option.  "`;

exports[`Vue TS Lint 正常通过 1`] = `""`;
